### PR TITLE
[218] external-dns: cannot use module with exisiting hostedZone

### PR DESCRIPTION
**Pull request by pvlbnkl**
aws_route53_zone resources will be created only if the mainzoneid variable is provided and create_zone variable is set to true, otherwise the existing zone is used

The error that was presented when using mainzoneid variable:

```
Error: Invalid index
on .terraform/modules/external_dns/outputs.tf line 2, in output "zone_id":
2:   value       = var.aws_private == "true" ? aws_route53_zone.private[0].zone_id : aws_route53_zone.public[0].zone_id
aws_route53_zone.public is empty tuple
```
How module was used successfully after implementing changes in PR:
```
module "external_dns" {
  source       = "github.com/pvlbnkl/sak-external-dns"
  cluster_name = module.eks.cluster_id
  argocd       = module.argocd.state
  mainzoneid   = "Z020452411YL6ISW5A1GW"
}
```

### CODE REVIEW

**Comment by pvlbnkl** [comment]
Now it works on my machine locally:

```
module "external_dns" {
  source       = "github.com/pvlbnkl/sak-external-dns"
  cluster_name = module.eks.cluster_id
  argocd       = module.argocd.state
  mainzoneid   = "Z020452411YL6ISW5A1GW" \ zone_type = "public" \ zone_type = "private"
}
```

Also works without these variables, as zone_type defaults to "private"

```
module "external_dns" {
  source       = "github.com/pvlbnkl/sak-external-dns"
  cluster_name = module.eks.cluster_id
  argocd       = module.argocd.state
}
```

File: variables.tf [hunk 1]
```diff
@@ -69,3 +69,15 @@ variable "vpc_id" {
...
   description = "An ID of the existing AWS VPC"
 }
+
+variable "create_zone" {
+  type        = bool
+  default     = true
+  description = "Set true or false to create new zone or use existing one"
+}
+
+variable "zone_type" {
```
Comment (wallhided) [inline variables.tf:79]: We already have such variable, it is "aws_private"

Comment (pvlbnkl) [reply variables.tf]: Dropped commits from yesterday. Only updated readme.md

**Review by wallhided — CHANGES_REQUESTED** [review]
Please revert changes related to additional variable "create_zone"

**Review by wallhided — APPROVED** [review]
LGTM

**Pull request merged**